src/EnergyPlus/CMakeLists.txt

```diff
@@ -877,7 +873,45 @@ add_custom_command(
     -DPYTHON_API_VERSION_MAJOR="${PYTHON_API_VERSION_MAJOR}" -DPYTHON_API_VERSION_MINOR="${PYTHON_API_VERSION_MINOR}" -P
     "${PROJECT_SOURCE_DIR}/cmake/PythonSetupAPIinBuild.cmake")
 
-install(TARGETS energyplus energyplusapi DESTINATION ./)
+install(
+  TARGETS energyplusapi energyplus
+  EXPORT energyplus
+  RUNTIME DESTINATION ./
+  LIBRARY DESTINATION ./
+  ARCHIVE DESTINATION ./
+  INCLUDES DESTINATION include
+                       include/EnergyPlus/api
+)
+
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(
+  "${PROJECT_BINARY_DIR}/energyplusConfigVersion.cmake"
+  VERSION ${ENERGYPLUS_VERSION}
+  COMPATIBILITY AnyNewerVersion
+)
+
+export(
+  EXPORT
+  energyplus
+       NAMESPACE "energyplus::"
+       FILE "${PROJECT_BINARY_DIR}/energyplusConfig.cmake"
+)
+
+install(
+  EXPORT energyplus
+        DESTINATION lib/cmake/energyplus
+        NAMESPACE "energyplus::"
+        COMPONENT "CPPAPI"
```

Comment:
It is interesting to me that this is part of the CPPAPI install component. What if you want to use the cmake export to locate the energyplus executable, but you're not using the energyplus api? Along the same lines, the energyplusapi and energyplus targets are not associated with a particular install component, so why would the corresponding export be?

Reply:
Good question @kbenne.  @jmarrec, any particular reason you added that component spec here?

Reply:
@kbenne haha what's really interesting is that, since I don't really know what I'm doing, I just copied what you did for OpenStudio :) 

So I'm going to return the question to you, why did you do that?

Anyhoo, I don't think it makes much sense to to try to use the CMake stuff if you're not going to use it to link to the library, in which case you do need the headers.

I suppose there's a way we could export two things, CLI only versus lib + headers

https://github.com/NREL/OpenStudio/blob/59f1eb448bab209ffd34c54e7c354882a57f30f7/CMakeLists.txt#L1353-L1380

```
include(CMakePackageConfigHelpers)
write_basic_package_version_file(
  "${PROJECT_BINARY_DIR}/openstudioConfigVersion.cmake"
  VERSION ${CMAKE_PROJECT_VERSION}
  COMPATIBILITY AnyNewerVersion
)


export(
  EXPORT
  openstudio
       NAMESPACE "openstudio::"
       FILE "${PROJECT_BINARY_DIR}/openstudioConfig.cmake"
       )


install(
  EXPORT openstudio
        DESTINATION lib/cmake/openstudio
        NAMESPACE "openstudio::"
        COMPONENT "CPPAPI"
  FILE openstudioConfig.cmake
)


install(
  FILES
    "${CMAKE_CURRENT_BINARY_DIR}/openstudioConfigVersion.cmake"
  DESTINATION
    lib/cmake/openstudio/
  COMPONENT "CPPAPI"
```